astropy/cosmology/funcs/tests/test_funcs.py

```diff
@@ -194,6 +198,11 @@ def test_z_at_value_bracketed(method):
     """
     cosmo = Planck13
 
+    if PYTEST_LT_8_0:
+        ctx = nullcontext()
+    else:
+        ctx = pytest.warns(AstropyUserWarning, match="fval is not bracketed")
+
     if method == "Bounded":
         with pytest.warns(AstropyUserWarning, match=r"fval is not bracketed"):
```

Comment:
This can be replaced with `ctx`

Reply:
No, because `ctx = nullcontext()` for `pytest < 8`.